state-chain/pallets/cf-swapping/src/tests/dca.rs

```diff
@@ -645,7 +626,17 @@ mod ccm_tests {
 	const GAS_SWAP_ID: SwapId = 3;
 
 	#[track_caller]
-	fn check_ccm_dca_request_received() {
+	fn setup_ccm_dca_swap(
+		number_of_chunks: u32,
+		chunk_interval: u32,
+		refund_params: Option<TestRefundParams>,
+	) {
```

Comment:
Similar to the other comment - why create two almost-but-not-quite identical functions setup_ccm_dca_swap and setup_dca_swap? Surely we want to do almost all the same checks? I can see differences between the two functions, but it's unclear if these differences are significant in any way.

Reply:
It got messy when accounting for the differences between ccm and not ccm, I would have to add logic to handle the gas and do extra checks. But I do not like putting logic in test code. It's better to have explicit hard coded asserts without any logic.